src/NodaTime/Calendars/WeekYearRule.cs

```diff
@@ -0,0 +1,199 @@
+        /// calendar year. Usually, some days of that week are in calendar year X, and some are in calendar year 
+        /// X-1. If <paramref name="minDaysInFirstWeek"/> or more of the days are in year X, then the week is
+        /// deemed to be the first week of week-year X. Otherwise, the week is deemed to be the last week of
+        /// week-year X-1, and the first week of week-year X starts on the following Monday.
+        /// </remarks>
+        /// <param name="minDaysInFirstWeek">The minimum number of days in the first Monday-to-Sunday week
+        /// which have to be in the new calendar year for that week to count as being in that week-year.
+        /// Must be in the range 1 to 7 inclusive.
+        /// </param>
+        /// <returns>A <see cref="WeekYearRule"/> with the specified minimum number of days in the first
+        /// week.</returns>
+        public static WeekYearRule ForMinDaysInFirstWeek(int minDaysInFirstWeek)
+        {
+            Preconditions.CheckArgumentRange(nameof(minDaysInFirstWeek), minDaysInFirstWeek, 1, 7);
+            return isoBasedRules[minDaysInFirstWeek - 1];
+        }
+
+        /// <summary>
+        /// Returns a <see cref="WeekYearRule"/> consistent with ISO-8601.
+        /// </summary>
+        /// <remarks></remarks>
+        /// <para>
+        /// In the standard ISO-8601 week algorithm, the first week of the year
+        /// is that in which at least 4 days are in the year. As a result of this
+        /// definition, day 1 of the first week may be in the previous year.
+        /// </para>
+        /// <para>
+        /// For example, January 1st 2011 was a Saturday, so only two days of that week
+        /// (Saturday and Sunday) were in 2011. Therefore January 1st is part of
+        /// week 52 of week-year 2010. Conversely, December 31st 2012 is a Monday,
+        /// so is part of week 1 of week-year 2013.
+        /// </para>
+        /// <value>A <see cref="WeekYearRule"/> consistent with ISO-8601.</value>
+        public static WeekYearRule Iso { get; } = isoBasedRules[3];
+
+        // TODO: public static WeekYearRule FromBclRule(CalendarWeekRule rule, IsoDayOfWeek firstDayOfWeek) => null;
+        // TODO: public static WeekYearRule FromBclRule(CalendarWeekRule rule, DayOfWeek firstDayOfWeek)
+        //    => FromBclRule(rule, BclConversions.ToIsoDayOfWeek(firstDayOfWeek));
+
+        // TODO: Make IsoBasedWeekYearRule more flexible, allowing non-Monday-to-Sunday weeks?
+
+        /// <summary>
+        /// Creates a <see cref="LocalDate" /> from a given week-year, week within that week-year,
+        /// and day-of-week, for the specified calendar system.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Valid values for <paramref name="weekYear"/> and <paramref name="weekOfWeekYear"/> will
+        /// depend on the rule and calendar system. Rules will usually permit any week-year which is also a valid calendar year,
```

Comment:
Perhaps "Usually, rules will permit...". I was expecting a sentence of the form "Rules will usually permit [...] _but_ [...]".